packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,473 @@
+      final Widget spacing = (widget.scrollDirection == Axis.vertical)
+          ? new SizedBox(height: _dropAreaExtent)
+          : new SizedBox(width: _dropAreaExtent);
+
+      // We open up a space under where the dragging widget currently is to
+      // show it can be dropped.
+      if (_currentIndex == index) {
+        return _buildContainerForScrollDirection(children: <Widget>[
+          new SizeTransition(
+            sizeFactor: _entranceController, 
+            axis: widget.scrollDirection,
+            child: spacing
+          ),
+          child,
+        ]);
+      }
+      // We close up the space under where the dragging widget previously was
+      // with the ghostController animation.
+      if (_ghostIndex == index) {
+        return _buildContainerForScrollDirection(children: <Widget>[
+          new SizeTransition(
+            sizeFactor: _ghostController, 
+            axis: widget.scrollDirection,
+            child: spacing,
+          ),
+          child,
+        ]);
+      }
+      return child;
+    }
+
+    // We wrap the drag target in a Builder so that we can scroll to its specific context.
+    return new Builder(builder: (BuildContext context) {
+      return new DragTarget<Key>(
+        builder: buildDragTarget,
+        onWillAccept: (Key toAccept) {
+          setState(() {
+            _nextIndex = index;
+            _requestAnimationToNextIndex();
+          });
+          _scrollTo(context);
+          // If the target is not the original starting point, then we will accept the drop.
+          return _dragging == toAccept && toAccept != toWrap.key;
+        },
+        onAccept: (Key accepted) {},
+        onLeave: (Key leaving) {},
+      );
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // We use the layout builder to constrain the cross-axis size of dragging child widgets.
+    return new LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
+        final List<Widget> wrappedChildren = <Widget>[];
+        for (int i=0; i<widget.children.length; i++) {
+          wrappedChildren.add(_wrap(widget.children[i], i, constraints));
+        }
+        wrappedChildren.add(_wrap(
+          new SizedBox(
+            height: widget.scrollDirection == Axis.horizontal 
+                ? constraints.maxHeight : 72.0, 
+            width: widget.scrollDirection == Axis.vertical 
+                ? constraints.maxWidth : 72.0,
+            key: const Key('DraggableList - End Widget'), 
+          ),
+          widget.children.length,
+          constraints),
+        );
+        return new SingleChildScrollView(
+          scrollDirection: widget.scrollDirection,
+          child: _buildContainerForScrollDirection(children: wrappedChildren),
+          padding: widget.padding, 
+          controller: _scrollController,
+        );
+    });
+  }
+}
+
+// Notifies its parent when its size has changed.
+class _SizeDetectingWidget extends StatefulWidget {
```

Comment:
@HansMuller, question:

Do you think this would be worth exposing at the framework level? It would be useful for people who want to compute things based on a widget's size.

This would be work done in a separate PR, of course.

Reply:
Long ago Flutter included a Widget like this called SizeObserver.  It was removed after it had spread like a virus :-). [Here's an explanation of why it was removed](https://groups.google.com/a/chromium.org/forum/#!original/blink-dev/iHXS9S66bAw/PuWbMRcWCAAJ).

Monitoring the widget's size in this way might be warranted in this case but I don't think we'll make it public.

Reply:
Acknowledged.